packages/hardhat-core/src/internal/cli/cli.ts

```diff
@@ -445,6 +452,91 @@ async function createNewProject() {
   throw new HardhatError(ERRORS.GENERAL.NOT_IN_INTERACTIVE_SHELL);
 }
 
+async function handleSecrets(args: string[]) {
```

Comment:
Let's move this to another file in the same directory, something like `internal/cli/secrets` or whatever.